packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Providers/TypeProvider.cs

```diff
@@ -539,25 +539,27 @@ private static bool IsMatch(MethodSignatureBase customMethod, MethodSignatureBas
             {
                 // The namespace may not be available for generated types as they are not yet generated
                 // so Roslyn will not have the namespace information.
-                if (string.IsNullOrEmpty(customMethod.Parameters[i].Type.Namespace))
+                if (!IsNameMatch(customMethod.Parameters[i].Type, method.Parameters[i].Type))
                 {
-                    if (customMethod.Parameters[i].Type.Name != method.Parameters[i].Type.Name)
-                    {
-                        return false;
-                    }
-                }
-                else
-                {
-                    if (customMethod.Parameters[i].Type.FullyQualifiedName != method.Parameters[i].Type.FullyQualifiedName)
-                    {
-                        return false;
-                    }
+                    return false;
                 }
             }
 
             return true;
         }
 
+        private static bool IsNameMatch(CSharpType typeFromCustomization, CSharpType generatedType)
+        {
+            // The namespace may not be available for generated types referenced from customization as they
+            // are not yet generated so Roslyn will not have the namespace information.
+            if (string.IsNullOrEmpty(typeFromCustomization.Namespace))
+            {
+                return typeFromCustomization.Name == generatedType.Name;
```

Comment:
lets use StringComparison.Ordinal on the name comparisons here

Reply:
Equality operator uses this under the covers - https://learn.microsoft.com/en-us/dotnet/api/system.string.op_equality?view=net-6.0#remarks